sdk/iot/core/src/az_iot_telemetry.c

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <stdint.h>
+
+#include "az_iot_hub_client.h"
+#include <az_contract_internal.h>
+#include <az_result.h>
+#include <az_span.h>
+
+static const uint8_t telemetry_prop_delim = '?';
+static const uint8_t telemetry_prop_separator = '&';
+static const uint8_t telemetry_null_terminator = '\0';
```

Comment:
I think the default behavior we are going with is a null terminated topic which can then be passed along to the MQTT library (important for those which require it to be null terminated. It will be documented as such and honestly if someone wants that extra byte back, we will get a GH issue and we can change it then. I don't want to complicate the compilation process early on if realistically it is unnecessary.